In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as st

In [3]:
data = pd.read_csv("Sales_add.csv")

In [4]:
data.head(10)

Month      Region      Manager  Sales_before_digital_add(in $)  \
0   Month-1  Region - A  Manager - A                          132921   
1   Month-2  Region - A  Manager - C                          149559   
2   Month-3  Region - B  Manager - A                          146278   
3   Month-4  Region - B  Manager - B                          152167   
4   Month-5  Region - C  Manager - B                          159525   
5   Month-6  Region - A  Manager - B                          137163   
6   Month-7  Region - C  Manager - C                          130625   
7   Month-8  Region - A  Manager - A                          131140   
8   Month-9  Region - B  Manager - C                          171259   
9  Month-10  Region - C  Manager - B                          141956   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402  
5                         256948  
6                         222106  
7                         230637  
8                         226261  
9                         193735

In [5]:
data.shape

(22, 5)

In [6]:
data.isna().sum()

Month                             0
Region                            0
Manager                           0
Sales_before_digital_add(in $)    0
Sales_After_digital_add(in $)     0
dtype: int64

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [8]:
data1 = data

In [9]:
data1.rename(columns= {"Sales_before_digital_add(in $)":"sales_bd","Sales_After_digital_add(in $)":"sales_ad"}, inplace=True)

#### 1. The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.


In [27]:
sales_bef = data1["sales_bd"]
sales_af = data1["sales_ad"]

Using one tailed T_test 
<br>
<br>
Null Hypothesis(H0): sales before stepping into digital marketing = sales after stepping into digital marketing
<br>
Alternate Hypothesis(H1): sales after stepping into digital marketing > sales before stepping into digital marketing
<br>
Level of significance(α) = 0.05

In [28]:
t_stat, p_value = st.ttest_ind(sales_af, sales_bef, alternative= "greater")

In [29]:
t_stat , p_value

(12.995084451110877, 1.3071840034523225e-16)

Since p_value < α, we can reject null hypothesis. 
<br>
So the sales after stepping into digital marketing has increased compared to sales before stepping into digital marketing.

#### 2. The company needs to check whether there is any dependency between the features “Region” and “Manager”.

Using Chi-square test for independence
<br>
<br>
Null Hypothesis(H0) = no dependency exist between "Region" and "Manager"
<br>
Alternate Hypothesis(H1) = Features "Region" and "Manager" are dependent
<br>
Level of significance (α)= 0.05

In [11]:
ch_data = data1[['Region', 'Manager']]

In [12]:
ch_data.head(5)

Region      Manager
0  Region - A  Manager - A
1  Region - A  Manager - C
2  Region - B  Manager - A
3  Region - B  Manager - B
4  Region - C  Manager - B

In [15]:
data1['Region'].value_counts()    #Counting categories in the feature "Region"

Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [18]:
data1['Manager'].value_counts()    #Counting categories in the feature "Manager"

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

In [19]:
obs = pd.crosstab(ch_data['Region'], ch_data['Manager'], margins=False)

In [20]:
obs

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [21]:
stat, p_value, dof, expected = st.chi2_contingency(obs)

In [23]:
stat, p_value, dof, expected

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

Since p(0.549) is greater than α , we cannot reject null hypothesis.
<br>
So there is no dependency between Managers and Region